packages/flutter/test/material/dropdown_test.dart

```diff
@@ -3052,17 +3052,180 @@ void main() {
     expect(itemBoxes[1].localToGlobal(Offset.zero).dy, 47.5);
 
     expect(
-      find.ancestor(
-        of: find.text(value).last,
-        matching: find.byType(CustomPaint),
-      ).at(2),
-      paints
-        ..save()
-        ..rrect()
-        ..rrect()
-        ..rrect()
+        find.ancestor(
+          of: find.text(value).last,
+          matching: find.byType(CustomPaint),
+        ).at(2),
+        paints
+          ..save()
+          ..rrect()
+          ..rrect()
+          ..rrect()
         // The height of menu is 47.0.
-        ..rrect(rrect: const RRect.fromLTRBXY(0.0, 0.0, 112.0, 47.0, 2.0, 2.0), color: Colors.grey[50], hasMaskFilter: false)
+          ..rrect(rrect: const RRect.fromLTRBXY(0.0, 0.0, 112.0, 47.0, 2.0, 2.0), color: Colors.grey[50], hasMaskFilter: false)
     );
   });
+
+  testWidgets('tapping a disabled item should not close DropdownButton', (WidgetTester tester) async {
+    final List<String> options = <String>['first', 'second', 'disabled'];
+    String? value = options.first;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: StatefulBuilder(
+            builder: (BuildContext context, StateSetter setState) => DropdownButton<String>(
+              value: value,
+              items: options.map((String s) => DropdownMenuItem<String>(
+                value: s,
+                disabled: s == 'disabled',
+                child: Text(s),
+              )).toList(),
+              onChanged: (String? v) => setState(() => value = v),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // Open dropdown.
+    await tester.tap(find.text('first').hitTestable());
+    await tester.pumpAndSettle();
+
+    // Tap on a disabled item.
+    await tester.tap(find.text('disabled').hitTestable());
+    await tester.pumpAndSettle();
+
+    // The dropdown should still be open, i.e., there should be one widget with 'second' text.
+    expect(find.text('second').hitTestable(), findsOneWidget);
+  });
+
+  testWidgets('disabled item should not be focusable when tapped', (WidgetTester tester) async {
```

Comment:
I think the correct test here would show that hovering over the disabled item (in web and desktop) keeps focus on original item.

The reason being that tapping on an (enabled) item does not focus the item, but instead closes the dropdown and selects it in the dropdown menu.

Reply:
I understand and agree with your point. Can you tell me how to do that? How do I simulate focusing / hovering over the disabled item in test.

Reply:
Here's a pointer to some hover tests. There are a bunch of others in the repo you can reference: https://github.com/flutter/flutter/blob/f835839f6b752a62a6f0463e51a64f20c4463982/packages/flutter/test/material/bottom_navigation_bar_test.dart#L1811

Reply:
I have added a test in the recently reviewed commit, to see if a disabled item is even focusable at all.
So should I add this hover test? Because if the disabled item is not within a focus (as verified by that test) it won't get the focus when hovered either.